pandas/plotting/_core.py

```diff
@@ -573,6 +573,8 @@ def _get_xticks(self, convert_period=False):
                 self.data = self.data[notna(self.data.index)]
                 self.data = self.data.sort_index()
                 x = self.data.index._mpl_repr()
+            elif all(isinstance(val, str) for val in index):
```

Comment:
I think the general preference is to use `pandas.compat.string_types` for `isinstance` checks against strings.  Looks like `string_types` is already imported, so just switching to `isinstance(val, string_types)` should be fine.

Reply:
Ok, I changed the line